src/ucp/core/ucp_request.inl

```diff
@@ -239,7 +239,7 @@ ucp_request_try_send(ucp_request_t *req, ucs_status_t *req_status,
         return 0;
     } else if (status != UCS_ERR_NO_RESOURCE) {
         /* Unexpected error */
-        ucp_request_complete_send(req, status);
+        /* request can be completed by user callback, can't be accessed anymore */
```

Comment:
ucp_request_send to return void, assert if status is err which is not UCS_INPROGRESS